powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -686,7 +687,7 @@ public InitActivationResponse initActivation(String applicationId, String userId
             // Get activation expiration date from request or from constants, if not provided
             Date timestampExpiration = activationExpireTimestamp;
             if (timestampExpiration == null) {
-                timestampExpiration = new Date(timestamp.getTime() + powerAuthServiceConfiguration.getActivationValidityBeforeActive());
+                timestampExpiration = Date.from(Instant.now().plus(powerAuthServiceConfiguration.getActivationValidityBeforeActive()));
```

Comment:
Are you sure that `timestamp` (before) is the same as `Instant.now()`? I think that there is a little time shift.